core/ts/Poll.ts

```diff
@@ -173,12 +177,30 @@ export class Poll implements IPoll {
   }
 
   /**
-   * Copy the state from the MaciState instance.
+   * Update a Poll with data from MaciState.
+   * This is the step where we copy the state from the MaciState instance,
+   * and set the number of signups we have so far.
    */
-  copyStateFromMaci = (): void => {
+  updatePoll = (numSignups: bigint): void => {
+    // there might be occasions where we fetch logs after new signups have been made
+    // logs are fetched (and MaciState/Poll created locally) after stateAq have been
+    // merged in. If someone signs up after that and we fetch that record
+    // then we won't be able to verify the processing on chain as the data will
+    // not match. For this, we must only copy up to the number of signups
+
     // Copy the state tree, ballot tree, state leaves, and ballot leaves
-    this.stateLeaves = this.maciStateRef.stateLeaves.map((x) => x.copy());
-    this.stateTree = this.maciStateRef.stateTree.copy();
+
+    // start by setting the number of signups
+    this.setNumSignups(numSignups);
+    // copy up to numSignups state leaves
+    this.stateLeaves = this.maciStateRef.stateLeaves.slice(0, Number(this.numSignups)).map((x) => x.copy());
```

Comment:
Is it ok to cast bigint to number? Just want to make sure if we don't have overflow here.

Reply:
so if we have `stateTreeDepth` at 10 we can have max `9765625` signups, at 20 we can have `95367431640625` signups and the maxSafeInt in JS is `9007199254740991` so we are good. Thinking though if it might be easier to just use a map for stateLeaves, and might be more efficient for the way we use the data

Reply:
I had a look and the changes would be substantial, we'd also need to update the merkle tree structure to use bigint indexing, so not sure it's worth the change. MaxSafeInt fits 1113716 the current world population so probably safe for now to keep using it. Lmk your thoughts